src/libs/ontuml2gufo/relation_functions.ts

```diff
@@ -0,0 +1,437 @@
+      quad(
+        namedNode(`:${uri}`),
+        namedNode('rdfs:comment'),
+        literal('Relation URI was automatically generated.'),
+      ),
+    );
+  }
+
+  return quads;
+}
+
+/**
+ * Transform relation cardinalities to gUFO
+ */
+function transformRelationCardinalities(
+  writer: N3Writer,
+  relation: IRelation,
+  options: IOntoUML2GUFOOptions,
+): Quad[] {
+  const { properties } = relation;
+
+  const domain = properties[0];
+  const range = properties[1];
+
+  const domainCardinality = domain.cardinality;
+  const rangeCardinality = range.cardinality;
+
+  if (domainCardinality && rangeCardinality) {
+    return [
+      ...transformRelationCardinality({
+        writer,
+        relation,
+        cardinality: rangeCardinality,
+        isDomain: true,
+        options,
+      }),
+      ...transformRelationCardinality({
+        writer,
+        relation,
+        cardinality: domainCardinality,
+        isDomain: false,
+        options,
+      }),
+    ];
+  }
+
+  return [];
+}
+
+/**
+ * Transform relation cardinalities to gUFO
+ */
+function transformRelationCardinality({
+  writer,
+  relation,
+  cardinality,
+  isDomain,
+  options,
+}: {
+  writer: N3Writer;
+  relation: IRelation;
+  cardinality: string;
+  isDomain: boolean;
+  options: IOntoUML2GUFOOptions;
+}): Quad[] {
+  // get domain
+  const classElement = isDomain ? relation.getSource() : relation.getTarget();
+  const classUri = getURI({
+    id: classElement.id,
+    name: classElement.name,
+    uriFormatBy: options.uriFormatBy,
+  });
+
+  const quads = [];
+
+  const lowerboundDomainCardinality = getLowerboundCardinality(cardinality);
+  const upperboundDomainCardinality = getUpperboundCardinality(cardinality);
+  const hasInfiniteCardinality = cardinality.includes('*');
+
+  // min = max
+  if (
+    lowerboundDomainCardinality === upperboundDomainCardinality &&
+    !hasInfiniteCardinality
+  ) {
+    quads.push(
+      generateRelationCardinalityQuad({
+        writer,
+        relation,
+        isDomain,
+        cardinalityPredicate: 'owl:qualifiedCardinality',
+        cardinality: lowerboundDomainCardinality,
+        options,
+      }),
+    );
+  }
+  // 0..3, 0..5, ...
+  else if (lowerboundDomainCardinality === 0 && !hasInfiniteCardinality) {
+    quads.push(
+      generateRelationCardinalityQuad({
+        writer,
+        relation,
+        isDomain,
+        cardinalityPredicate: 'owl:maxQualifiedCardinality',
+        cardinality: upperboundDomainCardinality,
+        options,
+      }),
+    );
+  }
+  // 2..*, 3..* ...
+  else if (
+    lowerboundDomainCardinality > 1 &&
+    lowerboundDomainCardinality < 99999 &&
+    hasInfiniteCardinality
+  ) {
+    quads.push(
+      generateRelationCardinalityQuad({
+        writer,
+        relation,
+        isDomain,
+        cardinalityPredicate: 'owl:minQualifiedCardinality',
+        cardinality: lowerboundDomainCardinality,
+        options,
+      }),
+    );
+  }
+  // 2..4, 3..7 ...
```

Comment:
This should include the 1...5 case as well.

Reply:
Done